frontend/web/components/SegmentOverrides.js

```diff
@@ -559,159 +559,178 @@ class TheComponent extends Component {
-                >
-                  Create Feature-Specific Segment
-                </Button>
-              </div>
-            )}
-          {this.props.showCreateSegment && !this.state.segmentEditId && (
-            <div className='create-segment-overrides'>
-              <CreateSegmentModal
-                onComplete={(segment) => {
-                  if (this.state.selectedSegment) {
-                    this.props.setShowCreateSegment(false)
-                  } else {
-                    this.props.setShowCreateSegment(false)
-                    this.setState(
-                      {
-                        selectedSegment: {
-                          label: segment.name,
-                          value: segment.id,
-                        },
-                      },
-                      this.addItem,
-                    )
-                  }
-                }}
-                onCancel={() => {
-                  this.props.setShowCreateSegment(false)
-                }}
-                condensed
-                feature={this.props.feature}
-                environmentId={this.props.environmentId}
-                projectId={this.props.projectId}
-              />
-            </div>
-          )}
-          {this.props.showCreateSegment && this.state.segmentEditId && (
-            <CreateSegmentModal
-              segment={this.state.segmentEditId}
-              isEdit
-              condensed
-              onComplete={() => {
-                this.setState({
-                  segmentEditId: undefined,
-                })
-                this.props.setShowCreateSegment(false)
-              }}
-              onCancel={() => {
-                this.setState({ segmentEditId: undefined })
-                this.props.setShowCreateSegment(false)
-              }}
-              environmentId={this.props.environmentId}
-              projectId={this.props.projectId}
-            />
-          )}
-          {visibleValues &&
-            !!visibleValues.length &&
-            !this.props.showCreateSegment && (
-              <div
-                style={isLoading ? { opacity: 0.5 } : null}
-                className='overflow-visible'
-              >
-                {!this.props.id && (
-                  <div className='my-4'>
-                    <InfoMessage className='mb-4 text-left faint'>
-                      Prioritise a segment override by dragging it to the top of
-                      the list.
-                      <br />
-                      Segment overrides will only apply when you identify via
-                      the SDK.{' '}
-                      <a
-                        target='_blank'
-                        href='https://docs.flagsmith.com/basic-features/managing-segments'
-                        rel='noreferrer'
+        <Permission
+          level='project'
+          permission={'MANAGE_SEGMENTS'}
+          id={this.props.projectId}
+        >
```

Comment:
Shouldn't this permission be `MANAGE_SEGMENT_OVERRIDES`

Reply:
I was following the advice you left in this comment. https://github.com/Flagsmith/flagsmith/issues/1676#issuecomment-1792703369

> In addition to the above, please also hide the Create Feature-Specific Segment button if the user is missing the project permission MANAGE_SEGMENTS since they will not be able to create a segment if that permission is missing.

I thought it made sense.

Reply:
Just in case, this button creates a Feature-Specific segment.

Reply:
Ahh, of course. Thanks for double checking Novak!